torch/distributed/distributed_c10d.py

```diff
@@ -1206,23 +1207,13 @@ def new_group(ranks=None, timeout=_default_pg_timeout):
     _check_default_pg()
 
     global _pg_group_ranks
-    global _group_count
-    global _pg_names
-
-    group_name = str(_group_count)
-    _group_count += 1
 
-    if group_name in _pg_names.values():
-        raise RuntimeError("The specified group name has already been "
-                           "created, please use a different group name")
-
-    default_backend, _ = _pg_map[_default_pg]
+    default_backend, default_store = _pg_map[_default_pg]
     global_rank = _default_pg.rank()
     global_world_size = _default_pg.size()
 
     # checks the input ranks
     if ranks is not None:
```

Comment:
What if different process provide the same set of ranks but in different order? Should we sort those ranks?

Reply:
I can see folks using a set to specify it, so added a call to `sorted(ranks)`.